sklearn/decomposition/dict_learning.py

```diff
@@ -132,9 +134,17 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         new_code = orthogonal_mp_gram(gram, cov, regularization, None,
                                       row_norms(X, squared=True),
                                       copy_Xy=copy_cov).T
+
+    elif algorithm == 'linear':
+        # Should be put into sparse_encode
+        # alpha = float(regularization) / n_features
+        lr = Ridge(alpha=0.0001 / n_features, fit_intercept=False, normalize=False)
```

Comment:
Why not use `LinearRegression`?  Is the ridge penalty beneficial in terms of speed or stability of the model? It seems to contradict the docstring.